lib/dns.js

```diff
@@ -61,17 +61,16 @@ function makeAsync(callback) {
       // The API already returned, we can invoke the callback immediately.
       callback.apply(null, arguments);
     } else {
-      process.nextTick(callMakeAsyncCbNT, callback, arguments);
+      var args = new Array(arguments.length + 1);
+      args[0] = callback;
+      for (var i = 1, a = 0; a < arguments.length; ++i, ++a)
+        args[i] = arguments[a];
+      process.nextTick.apply(null, args);
     }
   };
 }
```

Comment:
Yeah I was going to do that initially but I wanted see if I could write `dns` benchmarks to see what kind of difference it would make. Optimizing this could be a separate PR though I suppose...